File: android/src/main/kotlin/com/tqc/ads/flutter_admob_native_ads/layouts/Form6Builder.kt
package com.tqc.ads.flutter_admob_native_ads.layouts

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.view.Gravity
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.google.android.gms.ads.nativead.MediaView
import com.google.android.gms.ads.nativead.NativeAdView
import com.tqc.ads.flutter_admob_native_ads.styling.AdStyleManager
import com.tqc.ads.flutter_admob_native_ads.utils.DimensionUtils

/**
 * Form6 Builder - Vertical with Smaller Media (ad_6.png)
 * Layout: Column[Row[Icon + Column[Row[AD + Title] + Body]] + Media + CTA]
 */
object Form6Builder {

    fun build(context: Context, styleManager: AdStyleManager): NativeAdView {
        val nativeAdView = NativeAdView(context).apply {
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
        }

        // Main vertical container
        val mainContainer = LinearLayout(context).apply {
            orientation = LinearLayout.VERTICAL
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
            val containerPadding = styleManager.getContainerPaddingPx()
            setPadding(containerPadding[0], containerPadding[1], containerPadding[2], containerPadding[3])
        }
        styleManager.styleMainContainer(mainContainer)

        // Inner content container with background color
        val contentContainer = LinearLayout(context).apply {
            orientation = LinearLayout.VERTICAL
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
            background = GradientDrawable().apply {
                setColor(Color.parseColor("#F1E9E9"))
                cornerRadius = DimensionUtils.dpToPx(context, 8f).toFloat()
            }
            val padding = DimensionUtils.dpToPx(context, 8f)
            setPadding(padding, padding, padding, padding)
        }

        // Header row: Icon + Right column
        val headerRow = LinearLayout(context).apply {
            orientation = LinearLayout.HORIZONTAL
            gravity = Gravity.TOP
        }

        // Icon (Left)
        val iconView = ImageView(context).apply {
            scaleType = ImageView.ScaleType.CENTER_CROP
            val size = DimensionUtils.dpToPx(context, 42f)
              layoutParams = LinearLayout.LayoutParams(size, size).apply {
                marginEnd = DimensionUtils.dpToPx(context, 8f)
            }
            background = GradientDrawable().apply {
                setColor(Color.parseColor("#f0f0f0"))
                cornerRadius = DimensionUtils.dpToPx(context, 8f).toFloat()
            }
            clipToOutline = true
        }
        headerRow.addView(iconView)

        // Right column: Title row + Body
        val rightColumn = LinearLayout(context).apply {
            orientation = LinearLayout.VERTICAL
            layoutParams = LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1f)
        }

        // Title row with Ad label
        val titleRow = LinearLayout(context).apply {
            orientation = LinearLayout.HORIZONTAL
            gravity = Gravity.CENTER_VERTICAL
        }

        // Ad label
        val adLabel = TextView(context).apply {
            text = "Ad"
            textSize = 10f
            setTextColor(Color.parseColor("#5D4037"))
            setTypeface(null, android.graphics.Typeface.BOLD)
            background = GradientDrawable().apply {
                setColor(Color.parseColor("#FFE0B2"))
                cornerRadius = DimensionUtils.dpToPx(context, 3f).toFloat()
            }
            val padH = DimensionUtils.dpToPx(context, 6f)
            val padV = DimensionUtils.dpToPx(context, 2f)
            setPadding(padH, padV, padH, padV)
            layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            ).apply { marginEnd = DimensionUtils.dpToPx(context, 8f) }
        }
        titleRow.addView(adLabel)

        // Headline (takes remaining space)
        val headlineView = TextView(context).apply {
            textSize = 14f
            setTextColor(Color.BLACK)
            setTypeface(null, android.graphics.Typeface.BOLD)
            maxLines = 1
            ellipsize = android.text.TextUtils.TruncateAt.END
            layoutParams = LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1f)
        }
        titleRow.addView(headlineView)
        rightColumn.addView(titleRow)

        // Body text
        val bodyView = TextView(context).apply {
            textSize = 12f
            setTextColor(Color.parseColor("#757575"))
            maxLines = 1
            ellipsize = android.text.TextUtils.TruncateAt.END
            layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            ).apply { topMargin = DimensionUtils.dpToPx(context, 4f) }
        }
        rightColumn.addView(bodyView)

        headerRow.addView(rightColumn)
        contentContainer.addView(headerRow)

        // Media view (120dp height)
        val mediaView = MediaView(context).apply {
            layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                DimensionUtils.dpToPx(context, 120f)
            ).apply { topMargin = DimensionUtils.dpToPx(context, 12f) }
            background = GradientDrawable().apply {
                setColor(Color.parseColor("#f0f0f0"))
                cornerRadius = DimensionUtils.dpToPx(context, 8f).toFloat()
            }
            clipToOutline = true
        }
        contentContainer.addView(mediaView)

        // CTA Button
        val ctaButton = Button(context).apply {
            layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            ).apply { topMargin = DimensionUtils.dpToPx(context, 12f) }
        }
        styleManager.styleButton(ctaButton)
        contentContainer.addView(ctaButton)

        mainContainer.addView(contentContainer)
        nativeAdView.addView(mainContainer)
        nativeAdView.iconView = iconView
        nativeAdView.headlineView = headlineView
        nativeAdView.bodyView = bodyView
        nativeAdView.mediaView = mediaView
        nativeAdView.callToActionView = ctaButton

        return nativeAdView
    }
}

File: ios/Classes/Layouts/Form6Builder.swift
import UIKit
import GoogleMobileAds

/// Form6 Builder - Vertical Smaller Media (ad_6.png)
/// Layout: Column[Row[Icon + Column[Row[AD + Title] + Body]] + Media + CTA]
enum Form6Builder {
    static func build(styleManager: AdStyleManager) -> GADNativeAdView {
        let nativeAdView = GADNativeAdView()
        nativeAdView.translatesAutoresizingMaskIntoConstraints = false

        // Main vertical container
        let mainStack = UIStackView()
        mainStack.translatesAutoresizingMaskIntoConstraints = false
        mainStack.axis = .vertical
        mainStack.spacing = 12
        mainStack.alignment = .fill
        styleManager.styleMainContainer(mainStack)

        // Inner content container with background color
        let contentContainer = UIStackView()
        contentContainer.axis = .vertical
        contentContainer.spacing = 12
        contentContainer.alignment = .fill
        contentContainer.backgroundColor = UIColor(red: 0.945, green: 0.914, blue: 0.914, alpha: 1) // #F1E9E9
        contentContainer.layer.cornerRadius = 8
        contentContainer.clipsToBounds = true
        contentContainer.layoutMargins = UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 8)
        contentContainer.isLayoutMarginsRelativeArrangement = true

        // Header row: Icon + Right column
        let headerRow = UIStackView()
        headerRow.axis = .horizontal
        headerRow.spacing = 8
        headerRow.alignment = .top

        // Icon (Left)
        let iconView = UIImageView()
        iconView.contentMode = .scaleAspectFill
        iconView.backgroundColor = UIColor(white: 0.94, alpha: 1)
        iconView.layer.cornerRadius = 8
        iconView.clipsToBounds = true
        iconView.widthAnchor.constraint(equalToConstant: 42).isActive = true
        iconView.heightAnchor.constraint(equalToConstant: 42).isActive = true
        headerRow.addArrangedSubview(iconView)

        // Right column: Title row + Body
        let rightColumn = UIStackView()
        rightColumn.axis = .vertical
        rightColumn.spacing = 4
        rightColumn.alignment = .fill

        // Title row with Ad label
        let titleRow = UIStackView()
        titleRow.axis = .horizontal
        titleRow.spacing = 2
        titleRow.alignment = .center

        // Ad label container
        let adLabel = UILabel()
        adLabel.text = "AD"
        adLabel.font = .systemFont(ofSize: 8, weight: .bold)
        adLabel.textColor = .white
        adLabel.backgroundColor = .clear
        adLabel.textAlignment = .center

        let container = UIView()
        container.backgroundColor = UIColor(red: 0.996, green: 0.765, blue: 0.094, alpha: 1)
        container.layer.cornerRadius = 3
        container.clipsToBounds = true
        container.addSubview(adLabel)
        adLabel.translatesAutoresizingMaskIntoConstraints = false
        let padH: CGFloat = 6
        let padV: CGFloat = 2
        NSLayoutConstraint.activate([
            adLabel.topAnchor.constraint(equalTo: container.topAnchor, constant: padV),
            adLabel.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: -padV),
            adLabel.leadingAnchor.constraint(equalTo: container.leadingAnchor, constant: padH),
            adLabel.trailingAnchor.constraint(equalTo: container.trailingAnchor, constant: -padH),
        ])
        container.setContentHuggingPriority(.required, for: .horizontal)
        titleRow.addArrangedSubview(container)

        // Headline (takes remaining space)
        let headlineLabel = UILabel()
        headlineLabel.font = .systemFont(ofSize: 14, weight: .bold)
        headlineLabel.textColor = .black
        headlineLabel.numberOfLines = 2
        titleRow.addArrangedSubview(headlineLabel)
        rightColumn.addArrangedSubview(titleRow)

        // Body text
        let bodyLabel = UILabel()
        bodyLabel.font = .systemFont(ofSize: 12)
        bodyLabel.textColor = UIColor(white: 0.46, alpha: 1)
        bodyLabel.numberOfLines = 2
        rightColumn.addArrangedSubview(bodyLabel)

        headerRow.addArrangedSubview(rightColumn)
        contentContainer.addArrangedSubview(headerRow)

        // Media view (120dp height)
        let mediaView = GADMediaView()
        mediaView.backgroundColor = UIColor(white: 0.94, alpha: 1)
        mediaView.layer.cornerRadius = 8
        mediaView.clipsToBounds = true
        mediaView.heightAnchor.constraint(equalToConstant: 120).isActive = true
        contentContainer.addArrangedSubview(mediaView)

        // CTA Button
        let ctaButton = UIButton(type: .system)
        styleManager.styleButton(ctaButton)
        contentContainer.addArrangedSubview(ctaButton)

        mainStack.addArrangedSubview(contentContainer)
        nativeAdView.addSubview(mainStack)
        NSLayoutConstraint.activate([
            mainStack.topAnchor.constraint(equalTo: nativeAdView.topAnchor),
            mainStack.leadingAnchor.constraint(equalTo: nativeAdView.leadingAnchor),
            mainStack.trailingAnchor.constraint(equalTo: nativeAdView.trailingAnchor),
            mainStack.bottomAnchor.constraint(equalTo: nativeAdView.bottomAnchor)
        ])
        
        nativeAdView.iconView = iconView
        nativeAdView.headlineView = headlineLabel
        nativeAdView.bodyView = bodyLabel
        nativeAdView.mediaView = mediaView
        nativeAdView.callToActionView = ctaButton
        return nativeAdView
    }
}
